libs/utils/src/logging.rs

```diff
@@ -45,3 +45,89 @@ pub fn init(log_format: LogFormat) -> anyhow::Result<()> {
 
     Ok(())
 }
+
+/// Disable the default rust panic hook by using `set_hook`.
+///
+/// For neon binaries, the assumption is that tracing is configured before with [`init`], after
+/// that sentry is configured (if needed). sentry will install it's own on top of this, always
+/// processing the panic before we log it.
+///
+/// When the return value is dropped, the hook is reverted to std default hook (prints to stderr).
+/// If the assumptions about the initialization order are not held, use
+/// [`TracingPanicHookGuard::disarm`] but keep in mind, if tracing is stopped, then panics will be
+/// lost.
+#[must_use]
+pub fn replace_panic_hook_with_tracing_panic_hook() -> TracingPanicHookGuard {
+    std::panic::set_hook(Box::new(tracing_panic_hook));
+    TracingPanicHookGuard::new()
+}
+
+/// Drop guard which restores the std panic hook on drop.
+///
+/// Tracing should not be used when it's not configured, but we cannot really latch on to any
+/// imaginary lifetime of tracing.
+pub struct TracingPanicHookGuard {
+    act: bool,
+}
+
+impl TracingPanicHookGuard {
+    fn new() -> Self {
+        TracingPanicHookGuard { act: true }
+    }
+
+    pub fn disarm(&mut self) {
+        self.act = false;
+    }
+}
+
+impl Drop for TracingPanicHookGuard {
+    fn drop(&mut self) {
+        if self.act {
+            let _ = std::panic::take_hook();
+        }
+    }
```

Comment:
Re-posting my questions from a DM:

* Do we need `take_hook`? We could just leave it be there forever and nothing bad would happen, right? In that case, do we want to have a guard?
* Current API lets us call `replace_panic_hook_with_tracing_panic_hook` two and more times, which doesn't make sense. This may not be the 1st api with sharing this issue, but wouldn't it be better to have something like an OOP singleton?

Reply:
> Do we need take_hook?

Happy to remove this guard altogether, just added it to be in line with the rest of functionality for proxy binary.

It is not perfect, but it would claim it works if used correctly but no unsafety happens if misused, just the lack of panic logging and/or sentry logging.

> lets us call replace_panic_hook_with_tracing_panic_hook two and more times

Because each time `set_hook` will be used, the attempts will be idempotent.

I don't think this is a major issue, you can misuse many things in safe code and often you will not get correct results after misusing..

Should I just remove this?

Reply:
> Should I just remove the guard parts?

Looked into removing it, I think it will increase the incorrect behaviour area, but your call.

Reply:
Keeping it, even though it's impossible to make perfect. I don't think the multiple times use is interesting either. Resolving, but please re-raise.